pennylane/ops/op_math/sum.py

```diff
@@ -256,9 +267,13 @@ def diagonalizing_gates(self):
         Returns:
             list[.Operator] or None: a list of operators
         """
-
-        eigen_vectors = self.eigendecomposition["eigvec"]
-        return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        if self.overlapping_wires:
+            eigen_vectors = self.eigendecomposition["eigvec"]
+            return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        diag_gates = []
+        for summand in self.summands:
+            diag_gates.extend(summand.diagonalizing_gates())
+        return [qml.adjoint(gate) for gate in diag_gates]
```

Comment:
Let *O<sub>A</sub> = U<sub>A</sub>Λ<sub>A</sub>U†<sub>A</sub>* and *O<sub>B</sub> = U<sub>B</sub>Λ<sub>B</sub>U†<sub>B</sub>* be 2 operators that act on different Hilbert spaces.

Then:
*O<sub>A</sub> + O<sub>B</sub> = U<sub>A</sub>Λ<sub>A</sub>U†<sub>A</sub> + U<sub>B</sub>Λ<sub>B</sub>U†<sub>B</sub> = U<sub>A</sub>U<sub>B</sub> (Λ<sub>A</sub> + Λ<sub>B</sub>) U†<sub>A</sub>U†<sub>B</sub>*

Proof:
U<sub>A</sub>U<sub>B</sub> (Λ<sub>A</sub> + Λ<sub>B</sub>) U†<sub>A</sub>U†<sub>B</sub> = U<sub>A</sub>U<sub>B</sub>Λ<sub>A</sub>U†<sub>A</sub>U†<sub>B</sub> + U<sub>A</sub>U<sub>B</sub>Λ<sub>B</sub>U†<sub>A</sub>U†<sub>B</sub> = U<sub>A</sub>Λ<sub>A</sub>U†<sub>A</sub>U<sub>B</sub>U†<sub>B</sub> + U<sub>B</sub>Λ<sub>B</sub>U†<sub>B</sub>U<sub>A</sub>U†<sub>A</sub> = U<sub>A</sub>Λ<sub>A</sub>U†<sub>A</sub> + U<sub>B</sub>Λ<sub>B</sub>U†<sub>B</sub> = O<sub>A</sub> + O<sub>B</sub>

U<sub>A</sub>U<sub>B</sub> is the sum's diagonalising gate. The eigenvalues are the sum of the expanded eigenvalues of each summand.